### PR TITLE
Remove PHP 5, 7 references

### PR DESCRIPTION
**Pull request by saundefined**
Related to #555.

Some more, but I still keep looking :)

### CODE REVIEW

File: reference/opcache/book.xml [hunk 1]
```diff
@@ -13,9 +13,8 @@
    on each request.
   </para>
   <para>
-   This extension is bundled with PHP 5.5.0 and later, and is
-   <link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="&url.pecl.package;ZendOpcache">available in PECL</link>
-   for PHP versions 5.2, 5.3 and 5.4.
+   This extension is bundled with PHP, and is
+   <link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="&url.pecl.package;ZendOpcache">available in PECL</link>.
```
Comment (cmb69) [inline reference/opcache/book.xml:17]: Makes no sense to point to PECL (latest version from 2015).

File: reference/strings/functions/stristr.xml [hunk 2]
```diff
@@ -104,7 +104,7 @@
 <?php
   $email = 'USER@EXAMPLE.com';
   echo stristr($email, 'e'); // outputs ER@EXAMPLE.com
-  echo stristr($email, 'e', true); // As of PHP 5.3.0, outputs US
+  echo stristr($email, 'e', true); // US
```
Comment (cmb69) [inline reference/strings/functions/stristr.xml:107]: Maybe
```suggestion
  echo stristr($email, 'e', true); // outputs US
```

**Review by cmb69 — COMMENTED** [review]
Thank you! I left some comments. @Girgias may want to have a look, too.

File: reference/errorfunc/constants.xml [hunk 3]
```diff
@@ -226,27 +226,8 @@
       <constant>E_DEPRECATED</constant>, except it is generated in PHP code by
       using the PHP function <function>trigger_error</function>.
      </entry>
-     <entry>As of PHP 5.3.0</entry>
-    </row>
-
-    <row xml:id="errorfunc.constants.errorlevels.e-all">
```
[the remark this answers is not in the document]
Comment (saundefined) [reply reference/errorfunc/constants.xml:232]: Oops... overlooked 😥 Returned.

File: reference/mail/functions/mail.xml [hunk 4]
```diff
@@ -98,14 +98,6 @@ $text = str_replace("\n.", "\n..", $text);
...
        values are the respective header values.
       </para>
-      <note>
-       <para>
-        Before PHP 5.4.42 and 5.5.27, repectively, <parameter>additional_headers</parameter> did not have mail header
-        injection protection. Therefore, users must make sure specified headers
-        are safe and contains headers only. i.e. Never start mail body by putting
-        multiple newlines.
-       </para>
-      </note>
```
[the remark this answers is not in the document]
Comment (saundefined) [reply reference/mail/functions/mail.xml:108]: What do you think about it?

>PHP has mail header injection protection.
>If `additional_headers` contains non-safe or invalid headers
>(e.g. multiple newlines), PHP will not send the message.

[the remark this answers is not in the document]
Comment (cmb69) [reply reference/mail/functions/mail.xml:108]: Great! :)

**Review by Girgias — CHANGES_REQUESTED** [review]
I've only reviewed the files in the appendices folder but you've missed loads of cases:

`appendices/configure/servers.xml`:
  - Line 76: PHP 5.6 mention
 
 appendices/ini.core.xml:
  - Whole changelog section for `post_max_size`
  - Line 1640: `max_file_uploads` explanation has a mention of PHP 5
 
 appendices/reserved.xml:
  - Line 55
  - Line 140
  - Line 160
  - Line 180
  - Line 197
  - Line 242
  - Line 265
  - Line 268 (PHP 7.0.0 mention)
  - Line 284
  - Line 301
  - Line 304
  - The whole "Predefined classes as of PHP 5"/"Predefined interfaces and classes as of PHP 7" sections needs to be worked on (but this can be a separate PR)
  - Line 417 (Generator class)
  - (soft-) Reserved words sections, all the "as of PHP 7", can be dropped (lines 536-607)
  
I would rather that there is a full cleanup done folder by folder (or extension by extension when in /reference/) then all these spurious changes which are a PITA to deal with for translations.

I spent 3 hours yesterday on the French translation to try to bring it up to date but I can't just do a search for `PHP 5` to easily figure out where I need to go 99% of the time because some of the references might still be in the English version, an occasional miss is not the issue, but when at least 1/3 of the files modified still have references when it seems 3 commit have already touched the file. It begs the question how is this clean-up performed.

I sincerely appreciate the work done, but I'd rather have it done slowly and meticulously then what seems to be a shotgun approach.

**Comment by saundefined** [comment]
@Girgias, thank you, I convert PR into a draft and will continue to find PHP 5 entries.

Now the following questions have arisen:

What should I do with next pecl packages:
- [paradox](https://pecl.php.net/package/Paradox)
- [wincache](https://pecl.php.net/package/wincache)
- [mysqlnd_ms](https://pecl.php.net/package/mysqlnd_ms)
- [mysqlnd_qc](https://pecl.php.net/package/mysqlnd_qc)
- [ingres](https://pecl.php.net/package/ingres)

They're not maintained, is it reasonable to find in there?
Also in [mongo](https://www.php.net/manual/ru/book.mongo.php) and [mysql](https://github.com/php/doc-en/pull/551) extention.

[reference/spl/versions.xml](https://github.com/php/doc-en/blob/master/reference/spl/versions.xml#L614) line 614 — is it an error?
[language/oop5/changelog.xml](https://github.com/php/doc-en/blob/master/language/oop5/changelog.xml) — leave it as is?

Should I change code blocks like [this](https://github.com/php/doc-en/blob/master/features/commandline.xml#L789)?

**Comment by Girgias** [comment]
> @Girgias, thank you, I convert PR into a draft and will continue to find PHP 5 entries.
> 
> Now the following questions have arisen:
> 
> What should I do with next pecl packages:
> 
>     * [paradox](https://pecl.php.net/package/Paradox)
> 
>     * [wincache](https://pecl.php.net/package/wincache)
> 
>     * [mysqlnd_ms](https://pecl.php.net/package/mysqlnd_ms)
> 
>     * [mysqlnd_qc](https://pecl.php.net/package/mysqlnd_qc)
> 
>     * [ingres](https://pecl.php.net/package/ingres)
> 
> 
> They're not maintained, is it reasonable to find in there?
> Also in [mongo](https://www.php.net/manual/ru/book.mongo.php) and [mysql](https://github.com/php/doc-en/pull/551) extention.

PECL extensions which aren't maintained and which don't have a version running on PHP 7 should probably be deleted (so wincache would still be there as it has a PHP 7 release), but each removal should be it's own PR so that it can be discussed individually.

> [reference/spl/versions.xml](https://github.com/php/doc-en/blob/master/reference/spl/versions.xml#L614) line 614 — is it an error?

Looks like it is one, also most version.xml files need to be updated to include PHP 8, but that's a separate discussion.

> [language/oop5/changelog.xml](https://github.com/php/doc-en/blob/master/language/oop5/changelog.xml) — leave it as is?

It might make sense to move this to an appendix or something like that as a historical trace of language features.

> Should I change code blocks like [this](https://github.com/php/doc-en/blob/master/features/commandline.xml#L789)?

No, this is more noise than what is worth it IMHO.

File: appendices/ini.core.xml [hunk 5]
```diff
@@ -1643,8 +1615,6 @@ include_path = ".:${USER}/pear/php"
       <listitem>
        <para>
         The maximum number of files allowed to be uploaded simultaneously.
-        Starting with PHP 5.3.4, upload fields left blank on submission do not
-        count towards this limit.
```
Comment (Girgias) [inline appendices/ini.core.xml:1647]: This information is still relevant in PHP 7 and 8

File: appendices/tokens.xml [hunk 6]
```diff
@@ -19,10 +19,7 @@
   <para>
    T_* constants values are automatically generated based on PHP's underlying parser infrastructure.
    This means that the concrete value of a token may change between two PHP
-   versions. For example the <constant>T_FILE</constant> constant is
-   <literal>365</literal> in PHP 5.3, while the same value refers now to
-   <constant>T_TRAIT</constant> in PHP 5.4 and the value of <constant>T_FILE</constant>
-   is <literal>369</literal>. This means that your code should never rely directly
+   versions. This means that your code should never rely directly
```
Comment (Girgias) [inline appendices/tokens.xml:22]: Might be better to see if something similar occurred in PHP 8 and use that as an example

File: install/ini.xml [hunk 7]
```diff
@@ -194,50 +194,6 @@ $ PHP_INI_SCAN_DIR=/usr/local/etc/php.d: php
...
-        <entry>
-         The <varname>PHP_INI_SCAN_DIR</varname> environment variable can be
-         set to override the scan directory set via the configure script.
-        </entry>
-       </row>
-       <row>
-        <entry>5.1.0</entry>
-        <entry>
-         It is possible to refer to existing .ini variables from within .ini files.
-        </entry>
```
Comment (Girgias) [inline install/ini.xml:234]: Is this information still available somewhere else?

File: install/windows/index.xml [hunk 8]
```diff
@@ -43,11 +43,7 @@
     </para>
 
     <para>
-      PHP 5.5 and 5.6 require VC CRT 11 (Visual Studio 2012). See: <link xlink:href="https://www.microsoft.com/en-us/download/details.aspx?id=30679">https://www.microsoft.com/en-us/download/details.aspx?id=30679</link>
-    </para>
-
-    <para>
-      PHP 7.0 and 7.1 require VC CRT 14 (Visual Studio 2015).
+      PHP 7.1 requires VC CRT 14 (Visual Studio 2015).
```
Comment (Girgias) [inline install/windows/index.xml:46]:
```suggestion
      PHP 7.1 requires VC CRT 14 (Visual Studio 2015).
```
```suggestion
      PHP prior to version 7.2 requires VC CRT 14 (Visual Studio 2015).
```
Or don't drop the mention of PHP 7.0, as the current sentence makes it seem that PHP 7.0 doesn't require this

File: install/windows/manual.xml [hunk 9]
```diff
@@ -78,7 +78,7 @@ REM Configure FastCGI Variables
      <para>x86 - production use of PHP 5.5 and up.</para>
     </listitem>
     <listitem>
-     <para>x64 - production use of PHP 7 (unless its a 32-bit only version of Windows). PHP 5.5 and 5.6 x64 builds are experimental.</para>
+     <para>x64 - production use of PHP 7 (unless its a 32-bit only version of Windows).</para>
```
Comment (Girgias) [inline install/windows/manual.xml:81]:
```suggestion
     <para>x64 - production use of PHP 7 (unless its a 32-bit only version of Windows).</para>
```
```suggestion
     <para>x64 - production use (unless its a 32-bit only version of Windows).</para>
```
Same for section above, but @cmb69 does this mean 32bit windows is not support anymore as of PHP 7 or that one shouldn't use a 64bit binary on a 32bit system (which seems self explanatory)

File: reference/array/functions/list.xml [hunk 10]
```diff
@@ -25,16 +25,14 @@
...
     left-most parameter.
    </para>
    <para>
    If you are using plain variables, you don't have to worry about this. But
    if you are using arrays with indices you usually expect the order of the
    indices in the array the same you wrote in the <function>list</function>
-   from left to right, which is not the case in PHP 5, as it's assigned in the
-   reverse order.
+   from left to right.
    </para>
```
Comment (Girgias) [inline reference/array/functions/list.xml:36]: This whole section seems unnecessary now, as it highlights the difference in behviour between PHP 5 and 7

**Comment by Girgias** [comment]
I think it would be better to split this into multiple PRs each tackling one part of the manual, as this is becoming hard to review.

**Comment by saundefined** [comment]
Sure, then I close this one and split it

File: install/windows/manual.xml (hunk 9 above)
Comment (cmb69) [reply install/windows/manual.xml:81]: 32bit Windows is still supported; that info is related to PHP 5, which didn't have official production builds for 64bit Windows (these have been marked as experimental). Note that a few lines above there is still a mention of PHP 5.5.

Frankly, this list doesn't make much sense at all. It says there are four types of builds, and then lists ZTS, NTS, x86 and x64; altough the four types are ZTS-x86, NTS-x86, ZTS-x64 and NTS-x64. It also says that ZTS would be for single process Webservers, but it really is about multi-threaded environments.

Anyhow, to keep the PR focused:
````.xml
   <itemizedlist spacing="compact">
    <listitem>
     <para>Thread-Safe(TS) - for single process web servers, like Apache with mod_php</para>
    </listitem>
    <listitem>
     <para>Non-Thread-Safe(NTS) - for IIS and other FastCGI web servers (Apache with mod_fastcgi) and recommended for command-line scripts</para>
    </listitem>
    <listitem>
     <para>x86 - for 32bit and 64bit Windows</para>
    </listitem>
    <listitem>
     <para>x64 - recommended for 64bit Windows</para>
    </listitem>
   </itemizedlist>
````

**Comment by cmb69** [comment]
> Looks like it is one, also most version.xml files need to be updated to include PHP 8, but that's a separate discussion.

Note that there are still several [respective PRs](https://github.com/php/doc-en/pulls/mumumu) waiting for review.

**Pull request closed without merging**